File: clients/php/Model/Player.php
<?php

namespace Model {
    require_once 'Stream.php';

    /**
     * Game's participant (team of units)
     */
    class Player
    {
        /**
         * Unique id
         */
        public int $id;
        /**
         * Number of kills
         */
        public int $kills;
        /**
         * Total damage dealt to enemies
         */
        public float $damage;
        /**
         * Survival place (number of survivor teams currently/at the moment of death)
         */
        public int $place;
        /**
         * Team score
         */
        public float $score;
    
        function __construct(int $id, int $kills, float $damage, int $place, float $score)
        {
            $this->id = $id;
            $this->kills = $kills;
            $this->damage = $damage;
            $this->place = $place;
            $this->score = $score;
        }
    
        /**
         * Read Player from input stream
         */
        public static function readFrom(\InputStream $stream): Player
        {
            $id = $stream->readInt32();
            $kills = $stream->readInt32();
            $damage = $stream->readDouble();
            $place = $stream->readInt32();
            $score = $stream->readDouble();
            return new Player($id, $kills, $damage, $place, $score);
        }
        
        /**
         * Write Player to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32($this->id);
            $stream->writeInt32($this->kills);
            $stream->writeDouble($this->damage);
            $stream->writeInt32($this->place);
            $stream->writeDouble($this->score);
        }
    }
}

File: clients/rust/src/model/player.rs
use super::*;

/// Game's participant (team of units)
#[derive(Clone, Debug)]
pub struct Player {
    /// Unique id
    pub id: i32,
    /// Number of kills
    pub kills: i32,
    /// Total damage dealt to enemies
    pub damage: f64,
    /// Survival place (number of survivor teams currently/at the moment of death)
    pub place: i32,
    /// Team score
    pub score: f64,
}

impl trans::Trans for Player {
    fn write_to(&self, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        self.id.write_to(writer)?;
        self.kills.write_to(writer)?;
        self.damage.write_to(writer)?;
        self.place.write_to(writer)?;
        self.score.write_to(writer)?;
        Ok(())
    }
    fn read_from(reader: &mut dyn std::io::Read) -> std::io::Result<Self> {
        let id: i32 = trans::Trans::read_from(reader)?;
        let kills: i32 = trans::Trans::read_from(reader)?;
        let damage: f64 = trans::Trans::read_from(reader)?;
        let place: i32 = trans::Trans::read_from(reader)?;
        let score: f64 = trans::Trans::read_from(reader)?;
        Ok(Self {
            id,
            kills,
            damage,
            place,
            score,
        })
    }
}